vendor/github.com/docker/distribution/registry/client/blob_writer.go

```diff
@@ -99,7 +99,7 @@ func (hbu *httpBlobUpload) Write(p []byte) (n int, err error) {
 		return 0, fmt.Errorf("bad range format: %s", rng)
 	}
 
-	return (end - start + 1), nil
```

Comment:
lgtm :+1:

https://docs.docker.com/registry/spec/api/#blob-upload

> *On Success: Data Accepted*
>
> ```
> 204 No Content
> Location: /v2/<name>/blobs/uploads/<uuid>
> Range: 0-<offset>
> Content-Length: 0
> Docker-Upload-UUID: <uuid>
> ```
>
> `Range` Range indicating the current progress of the upload.

So, basically, (end - start + 1) is a total size of uploaded data, and this is not what the writer should return.